client/signup/utils.js

```diff
@@ -93,6 +95,27 @@ function getNextStepName( flowName, currentStepName ) {
 	return flow.steps[ indexOf( flow.steps, currentStepName ) + 1 ];
 }
 
+function getValueFromProgressStore( { signupProgressStore, stepName, fieldName } ) {
+	if ( ! signupProgressStore ) {
```

Comment:
I think we should move the blocks that check for the presence of the progress store out of these two utility functions and into `SignupForm`. The main reason for this is that it will make the behavior of `SignupForm` (which must behave with and without the `signupProgressStore` prop) more explicit if we skip `autoFillUsername` altogether when that component isn't provided `signupProgressStore`.